### DISCUSSION TITLE
How can I tell what collection a user is logged in from and whether they're an admin or not?

### DISCUSSION THREAD

**Discussion by mrvillage** [pocketbase/pocketbase#1109] [Q&A] [answered]
The question basically says it all, if I have a logged in client how can I tell if the user is from a normal auth collection like `users` or maybe something like `managers` or if they're an admin and which collection exactly they're from? Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
It depends how you are using PocketBase and where you want to perform the check.

Admins and records are 2 different entities and they don't share the same representation. **Collection API rules don't apply for admins and they can access everything.**

Each auth record has `collectionId` and `collectionName` props. They can be used also as part of the auth filter, eg. `@request.auth.collectionName = "users"`.

---

If you want to get the current auth model type in the SDKs, you can check the object instance, eg. for the JS SDK:

```js
import PocketBase, { Record, Admin } from pocketbase;

const pb = new PocketBase("http://127.0.0.1:8090");

...

if (pb.authStore.model instanceof Admin) {
    console.log("admin...")
} else if (pb.authStore.model instanceof Record) {
    console.log("auth record...")
} else {
    console.log("guest...")
}
```

---

If you use PocketBase as framework you could check https://pocketbase.io/docs/router/#get-logged-admin-or-app-user.

**Reply by mrvillage** [reply]
That's perfect! I was trying to figure it out on the client side with the SDK, I didn't realize there was a special class. Thank you so much!!!

**Comment by mrbu-solutions** [comment]
Same question and (pb.authStore.model instanceof Admin) works fine.

**Comment by Qa5imm** [comment]
Has this been changed? I'm trying to get Admin from pockectbase and using it as above but getting an error
Right-hand side of 'instanceof' is not an object. I tried console.log Admin and it gives me undefined

**Reply by ganigeorgiev** [reply]
@Qa5imm Yes, with JS SDK v0.17+ there are now dedicated `pb.authStore.isAdmin` and `pb.authStore.isAuthRecord` fields. For all available fields please check https://github.com/pocketbase/js-sdk#common-auth-store-fields-and-methods.

**Reply by Qa5imm** [reply]
thank youu!!
